### ISSUE TITLE
`

### ISSUE THREAD

**Issue by tosincarik** [tosincarik/Tee-Data-Portfolio#1]
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

!pip  install opendatasets
!pip install pandas

!pip install kaggle

mkdir ~/.kaggle

cp kaggle.json ~/.kaggle/

import opendatasets as od
import pandas

! chmod 600 ~/.kaggle/kaggle.json

! kaggle datasets download akshaydattatraykhare/data-for-admission-in-the-university

! unzip data-for-admission-in-the-university

df_adm = pd.read_csv('adm_data.csv')

## Check if there are any missing values in the dataset

df_adm.isnull().values.any()

df_adm.head()

df_adm.describe()

df_adm.info()

## Split the dataset and fetch the independent variables into X
X = df_adm.iloc[:, 1:-1].values
print(X)

## Split the dataset to fetch the dependent variables into y
y = df_adm.iloc[:, -1].values
print(y)

#Split the dataset into training and test sets
#Using scikit learn library , import the train test
from sklearn.model_selection import train_test_split
X_train, X_test,y_train,y_test = train_test_split(X,y, test_size=0.2, random_state=0)

print(X_test)

### **Multiple Linear Regression**

#Import the linearregression object and create an instance of the object
#Train the multiple linear regression on the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

##Predictor for Linear Regression

y_pred = regressor.predict(X_test)

print(y_pred)

print(y_test)

# Comparing the results of the predicted chance of admit with the actual(real) chance of admit test set
# Set to display only in two decimal places
np.set_printoptions(precision=2)
print(np.concatenate((y_pred.reshape(len(y_pred),1) , y_test.reshape(len(y_test),1)),1))

##Accuracy Score

from sklearn.metrics import r2_score
r2_score(y_pred,y_test)

## Polynomial Regression

#convert the matrix of features of X to a polynomial matrix
from sklearn.preprocessing import PolynomialFeatures
pol_reg = PolynomialFeatures(degree=3)
pol_reg.fit_transform(X)
X_Pol = pol_reg.fit_transform(X)
pol_reg2 = LinearRegression()
pol_reg2.fit(X_Pol,y)

##Predictor for PolyRegression

y_pred = pol_reg2.predict(X_Pol)

# Comparing the results of the predicted chance of admit with the actual(real) chance of admit test set
# Set to display only in two decimal places
np.set_printoptions(precision=2)
print(np.concatenate((y_pred.reshape(len(y_pred),1) , y.reshape(len(y),1)),1))

from sklearn.metrics import r2_score
r2_score(y,y_pred)
